tests/cupy_tests/linalg_tests/test_eigenvalue.py

```diff
@@ -27,14 +27,32 @@ def _get_hermitian(xp, a, UPLO):
     reason='eigensolver not added until ROCm 4.2.0')
 class TestEigenvalue:
 
-    @testing.for_all_dtypes(no_float16=True, no_complex=True)
+    @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-4, contiguous_check=False)
     def test_eigh(self, xp, dtype):
-        a = xp.array([[1, 0, 3], [0, 5, 0], [7, 0, 9]], dtype)
+        if xp == numpy and dtype == numpy.float16:
+            # NumPy's eigh does not support float16
+            _dtype = 'f'
+        else:
+            _dtype = dtype
+        a = xp.array([[1, 0, 3], [0, 5, 0], [7, 0, 9]], _dtype)
         w, v = xp.linalg.eigh(a, UPLO=self.UPLO)
-        # NumPy, cuSOLVER, rocSOLVER all sort in ascending order,
-        # so they should be directly comparable
-        return w, v
+
+        # Changed the verification method to check if Av and vw match, since
+        # the eigenvectors of eigh() with CUDA 11.6 are mathematically correct
+        # but may not match NumPy.
+        if self.UPLO == 'U':
+            A = xp.triu(a) + xp.transpose(xp.triu(a, 1))
+        else:
+            A = xp.tril(a) + xp.transpose(xp.tril(a, -1))
+        if dtype == numpy.float16:
+            atol = 1e-3
+        else:
+            atol = 1e-5
+        testing.assert_allclose(A @ v, v @ xp.diag(w), atol=atol, rtol=1)
```

Comment:
`rtol=1` is quite large.  Is there any problem with configuring as `atol=tol, rtol=tol`?

Reply:
I also think that `rtol=1` is undesirable.
However, in this test case, there is a part where `Av` and `vw` are almost zero, and due to the rounding error of floating point, it doesn't seem to work well when verifying with relative error.

Reply:
I see that error may be large near zero in the check.  Then I would use `rtol=0` and larger `atol`.

Reply:
When I tried it before, an error occurred when the `rtol` value was small, but when I set the `rtol` value to be the same as the `atol` value again, no error occurred. I'm not sure what was happening, but for now, I'll revert the code so that it sets the `rtol` and `atol` values to be the same.